src/python/sdk/econia_sdk/view/market.py

```diff
@@ -320,6 +320,178 @@ def get_price_levels_all(view: EconiaViewer, market_id: int) -> dict:
+            len(bids) >= max_bids or next_bid_idx == "0"
+        ):
+            keep_looping = False
+        elif next_ask_idx == "0" or next_bid_idx == "0":
+            if next_ask_idx == "0":
+                ask_page_size = 0
+            else:
+                bid_page_size = 0
+
+    return {"asks": asks, "bids": bids, "market_id": market_id}
+
+
+def get_price_levels_paginated(
+    viewer: EconiaViewer,
+    market_id: int,
+    ask_page_size: int,
+    bid_page_size: int,
+    next_ask_idx: int,
+    next_bid_idx: int,
+    ledger_version: int,
+):
+    returns = viewer.get_returns(
+        "market",
+        "get_price_levels_paginated",
+        [],
+        [
+            str(market_id),
+            str(ask_page_size),
+            str(bid_page_size),
+            str(next_ask_idx),
+            str(next_bid_idx),
+        ],
+        ledger_version=ledger_version,
+    )
+    return returns
+
+
+def get_price_levels_with_pagination(
+    viewer: EconiaViewer,
+    market_id: int,
+    max_asks: int = 100,
+    max_bids: int = 100,
+    page_size: int = 100,
+) -> dict:
+    info = viewer.aptos_client.info()
+    ledger_version = int(info["ledger_version"])
+
+    next_ask_idx = "0"
+    next_bid_idx = "0"
+    bid_page_size = ask_page_size = page_size
+
+    asks = dict()
+    asks_size = 0
+    bids = dict()
+    bids_size = 0
+
+    keep_looping = True
+    while keep_looping:
+        returns = get_price_levels_paginated(
+            viewer,
+            market_id,
+            ask_page_size,
+            bid_page_size,
+            int(next_ask_idx),
+            int(next_bid_idx),
+            ledger_version,
+        )
+
+        value = returns[0]
+        next_ask_idx = returns[1]
+        next_bid_idx = returns[2]
+
+        for ask in value["asks"]:
+            price = int(ask["price"])
+            size = int(ask["size"])
+            if price in asks:
+                asks[price] += size
+            else:
+                asks[price] = size
+        for bid in value["bids"]:
+            price = int(bid["price"])
+            size = int(bid["size"])
+            if price in bids:
+                bids[price] += size
+            else:
+                bids[price] = size
+        asks_size += ask_page_size
+        bids_size += bid_page_size
+
+        if (asks_size >= max_asks or next_ask_idx == "0") and (
+            bids_size >= max_bids or next_bid_idx == "0"
+        ):
+            keep_looping = False
+        elif next_ask_idx == "0" or next_bid_idx == "0":
+            if next_ask_idx == "0":
+                ask_page_size = 0
+            else:
+                bid_page_size = 0
+
+    asks_ls = []
+    bids_ls = []
+    for price, size in asks.items():
+        asks_ls.append({"price": price, "size": size})
+    for price, size in bids.items():
+        bids_ls.append({"price": price, "size": size})
+    return {"asks": asks_ls, "bids": bids_ls, "market_id": market_id}
```

Comment:
Is the dict `items()` method sorted?

If not, which I suspect it isn't, then how about sorting by keys (ascending for asks, descending for bids) at some point in this operation?

Also, I'm guessing this can probably be done in a one-liner using a list/dict comprehesion or a zip operator

Reply:
Python dictionaries are sorted by item insertion, and even if an item is updated its place in the ordering remains the same.